cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java

```diff
@@ -185,39 +186,48 @@ private static KafkaConnectS2IAssemblyOperator createS2iOperator(Vertx vertx, Ku
             OkHttpClient ok = client.adapt(OkHttpClient.class);
-            Future<Boolean> fut = Future.future();
+            Future<PlatformFeaturesAvailability> fut = Future.future();
 
             vertx.executeBlocking(request -> {
-                try (Response resp = ok.newCall(new Request.Builder().get().url(client.getMasterUrl().toString() + "apis/route.openshift.io/v1").build()).execute()) {
-                    if (resp.code() >= 200 && resp.code() < 300) {
-                        log.debug("{} returned {}. We are on OpenShift.", resp.request().url(), resp.code());
-                        // We should be on OpenShift based on the /apis/route.openshift.io/v1 result. We can now safely try isAdaptable() to be 100% sure.
-                        Boolean isOpenShift = Boolean.TRUE.equals(client.isAdaptable(OpenShiftClient.class));
-                        request.complete(isOpenShift);
+                try {
+                    Boolean isOpenShift;
+                    if (config.isAssumeOpenShift() != null)  {
+                        log.debug("OpenShift has been set to {} through {}.", config.isAssumeOpenShift(), ClusterOperatorConfig.STRIMZI_ASSUME_OPENSHIFT);
+                        isOpenShift = config.isAssumeOpenShift();
                     } else {
-                        log.debug("{} returned {}. We are not on OpenShift.", resp.request().url(), resp.code());
-                        request.complete(Boolean.FALSE);
+                        Response resp = ok.newCall(new Request.Builder().get().url(client.getMasterUrl().toString() + "apis/route.openshift.io/v1").build()).execute();
+                        if (resp.code() >= 200 && resp.code() < 300) {
+                            log.debug("{} returned {}. We are on OpenShift.", resp.request().url(), resp.code());
+                            // We should be on OpenShift based on the /apis/route.openshift.io/v1 result. We can now safely try isAdaptable() to be 100% sure.
+                            isOpenShift = Boolean.TRUE.equals(client.isAdaptable(OpenShiftClient.class));
+                        } else {
+                            log.debug("{} returned {}. We are not on OpenShift.", resp.request().url(), resp.code());
+                            isOpenShift = false;
+                        }
                     }
+                    /* test */ String major = client.getVersion().getMajor().equals("") ? "1" : client.getVersion().getMajor();
+                    /* test */ String minor = client.getVersion().getMinor().equals("") ? "9" : client.getVersion().getMinor();
+
+                    PlatformFeaturesAvailability pfa = new PlatformFeaturesAvailability(isOpenShift,
+                            new KubernetesVersion(Integer.parseInt(major.split("\\D")[0]),
+                                    Integer.parseInt(minor.split("\\D")[0])));
+                    request.complete(pfa);
                 } catch (IOException e) {
```

Comment:
We need to complete the future for `RuntimeExceptions` too, so just catch `Exception` here.